pkg/roles/utils.go

```diff
@@ -58,3 +58,29 @@ func GetHcpSharedVpcPolicyDetails(r *rosa.Runtime, roleArn string) (bool, string
 
 	return existsQuery != nil, createPolicy, policyName, nil
 }
+
+func CheckIfRolesAreHcpSharedVpc(r *rosa.Runtime, roles []string) bool {
+	isHcpSharedVpc := false
+	for _, roleName := range roles {
+		ptrRoleName := roleName
+		attachedPolicies, err := r.AWSClient.GetPolicyDetailsFromRole(&ptrRoleName)
+		if err != nil {
+			r.Reporter.Errorf("Failed to get policy details for role '%s': %v", roleName, err)
+		}
+		for _, attachedPolicy := range attachedPolicies {
+			rhManaged := false
```

Comment:
Might be worth making lines 71 - 79 in to a helper func since its being duplicated in the follow up draft pr